src/utils/api.ts

```diff
@@ -1,20 +1,22 @@
 import { Song, SongFromAPI } from '../types';
 
-export async function getSong(artistName?: string): Promise<Song[]> {
+export async function getSongs(artistName?: string): Promise<Song[]> {
   const response = await fetch(
     `https://www.songsterr.com/a/ra/songs.json?pattern=${
       artistName ? artistName : ''
     }`
   );
   const data: SongFromAPI[] = await response.json();
-  const songs: Song[] = data.map((song) => {
-    const formattedSong: Song = {
-      id: song.id,
-      artistName: song.artist.name,
-      songName: song.title,
-    };
-    return formattedSong;
+  const songs: Song[] = data.map((song, index) => {
+    if (index < 11) {
+      const formattedSong: Song = {
+        id: song.id,
+        artistName: song.artist.name,
+        songName: song.title,
+      };
+      return formattedSong;
+    }
+    return;
```

Comment:
In your if statement (it's not a "Schleife" :)) is a case that will return an "undefined" which is not acceptable for your type `Song[]`

Reply:
I create a new Error object, if formatted song don't exist but  that still doesn't work